alacritty_terminal/src/ansi.rs

```diff
@@ -750,10 +750,17 @@ where
             // Set window title
             b"0" | b"2" => {
                 if params.len() >= 2 {
-                    if let Ok(utf8_title) = str::from_utf8(params[1]) {
-                        self.handler.set_title(utf8_title);
-                        return;
+                    let mut buf = String::new();
+                    if let Ok(title_segment) = str::from_utf8(params[1]) {
+                        buf.push_str(title_segment);
+                    }
+                    for item in &params[2..] {
+                        if let Ok(title_segment) = str::from_utf8(item) {
+                            buf.push(';');
+                            buf.push_str(title_segment);
+                        }
                     }
+                    self.handler.set_title(&buf);
                 }
                 unhandled(params);
```

Comment:
I think we can simplify this code significantly:

```rust
                if params.len() >= 2 {
                    let title = params[1..]
                        .iter()
                        .flat_map(str::from_utf8)
                        .collect::<Vec<&str>>()
                        .join(";");
                    self.handler.set_title(&title);
                    return;
                }
                unhandled(params);
```

Reply:
Yes, that looks nicer. However, I've had to eta-expand the argument to `flat_map` because of some error with lifetimes I didn't totally understand.

Reply:
Oh yeah, sorry. I meant to post it with a dedicated closure since the types of the functions don't match up entirely, looks like I forgot to edit that, your version is exactly what I was going for!